src/kubectlutils.ts

```diff
@@ -17,157 +17,126 @@
 import * as k8s from 'vscode-kubernetes-tools-api';
 import * as config from './config';
 
-export function isKubernetesAvailable(): Promise<boolean> {
-	return new Promise<boolean>( (resolve) => {
-		k8s.extension.kubectl.v1
-			.then( (kubectl) => {
-				resolve(kubectl && kubectl.available);
-			})
-			.catch( err => {
-				resolve(false);
-			});
-	});
+export async function isKubernetesAvailable(): Promise<boolean> {
+	const kubectl: k8s.API<k8s.KubectlV1> = await k8s.extension.kubectl.v1;
+	return kubectl && kubectl.available;
 }
 
-export function getNamedListFromKubernetes(itemType : string, extra? : string): Promise<string> {
-	return new Promise<string>( async (resolve, reject) => {
-		await k8s.extension.kubectl.v1
-			.then( async (kubectl) => {
-				let cmd = `get ${itemType}`;
-				if (extra) {
-					cmd += ` ${extra}`;
-				}
-				if (kubectl && kubectl.available) {
-					return await kubectl.api.invokeCommand(cmd);
-				} else {
-					reject(new Error('Kubernetes not available'));
-				}
-			})
-			.then( (result) => {
-				if (!result || result.code !== 0) {
-					let error = `Unable to invoke kubectl to retrieve ${itemType}`;
-					if (result && result.stderr) {
-						error = result.stderr;
-					}  
-					reject(error);
-				} else if (result) {
-					const splitResults = result.stdout;
-					resolve(splitResults);
-				}
-			})
-			.catch( (err) => reject(err) );
-	});
+export async function getNamedListFromKubernetes(itemType : string, extra? : string): Promise<string> {
+	const kubectl: k8s.API<k8s.KubectlV1> = await k8s.extension.kubectl.v1;
+	let cmd = `get ${itemType}`;
+	if (extra) {
+		cmd += ` ${extra}`;
+	}
+	if (kubectl && kubectl.available) {
+		const result: k8s.KubectlV1.ShellResult | undefined = await kubectl.api.invokeCommand(cmd);
+		if (!result || result.code !== 0) {
+			let error = `Unable to invoke kubectl to retrieve ${itemType}`;
+			if (result && result.stderr) {
+				error = result.stderr;
+			}  
+			return error;
+		} else if (result) {
+			const splitResults = result.stdout;
+			return splitResults;
+		}
+	}
+	return 'Kubernetes not available';
```

Comment:
Isn't `kubectl` which is not available?

Reply:
I just took what was there before, but you are right.

Reply:
If kubectl is unavailable, it likely means that the Kubernetes tools extension is unavailable (thus the k8s api). Though it may be more correct to say that 'Kubernetes Tools API is not available' -- to our extension, it amounts to the same thing. We are unable to see into a Kubernetes system because the kubectl API isn't there. Is the user really going to care one way or the other?

Reply:
Probably not....it doesn't work in that case...that is all he cares about.

Reply:
it can be OpenShift too.

if what is missing is kubectl, we should tell them that kubectl is missing not that something else is missing.
Even without kubectl, some kamel commands can work and reach the Kubernetes/OpenShift instance.